modules/widgets/authors.php

```diff
@@ -0,0 +1,180 @@
+		wp_cache_delete( 'widget_authors_ssl', 'widget' );
+	}
+
+	public function widget( $args, $instance ) {
+		global $wpdb;
+
+		$cache_bucket = is_ssl() ? 'widget_authors_ssl' : 'widget_authors';
+
+		if ( '%BEG_OF_TITLE%' != $args['before_title'] ) {
+			if ( $output = wp_cache_get( $cache_bucket, 'widget') ) {
+				echo $output;
+				return;
+			}
+
+			ob_start();
+		}
+
+		$instance = wp_parse_args( $instance, array( 'title' => __( 'Authors' ), 'all' => false, 'number' => 5, 'avatar_size' => 48 ) );
+		$instance['number'] = min( 10, max( 0, (int) $instance['number'] ) );
+
+		// We need to query at least one post to determine whether an author has written any posts or not
+		$query_number = max( $instance['number'], 1 );
+
+		$authors = get_users( array(
+			'fields' => 'all',
+			'who' => 'authors'
+		) );
+
+		echo $args['before_widget'];
+		echo $args['before_title'] . esc_html( $instance['title'] ) . $args['after_title'];
+		echo '<ul>';
+
+		foreach ( $authors as $author ) {
+			$r = new WP_Query( array(
+				'author'         => $author->ID,
+				'posts_per_page' => $query_number,
+				'post_type'      => 'post',
+				'post_status'    => 'publish',
+				'no_found_rows'  => true,
+			) );
+
+			if ( ! $r->have_posts() && ! $instance['all'] )
+				continue;
+
+			echo '<li>';
+
+			// Display avatar and author name
+			if ( $r->have_posts() ) {
+				echo '<a href="' . get_author_posts_url( $author->ID ) . '">';
+
+				if ( $instance['avatar_size'] > 1 )
+					echo ' ' . get_avatar( $author->ID, $instance['avatar_size'], '', true ) . ' ';
+
+				echo '<strong>' . esc_html( $author->display_name ) . '</strong>';
+				echo '</a>';
+			}
+			else if ( $instance['all'] ) {
+				if ( $instance['avatar_size'] > 1 )
+					echo get_avatar( $author->ID, $instance['avatar_size'], '', true ) . ' ';
+
+				echo '<strong>' . esc_html( $author->display_name ) . '</strong>';
+			}
+
+			if ( 0 == $instance['number'] ) {
+				echo '</li>';
+				continue;
+			}
+
+			// Display a short list of recent posts for this author
+
+
+			if ( $r->have_posts() ){
+				echo '<ul>';
+
+				while ( $r->have_posts() ) {
+					$r->the_post();
+					echo '<li><a href="' . get_permalink() . '">';
+
+					if ( get_the_title() )
+						echo get_the_title();
+					else
+						echo get_the_ID();
+
+					echo '</a></li>';
+				}
+
+				echo '</ul>';
+			}
+
+			echo '</li>';
+		}
+
+		echo '</ul>';
+		echo $args['after_widget'];
+
+		wp_reset_postdata();
+
+		if ( '%BEG_OF_TITLE%' != $args['before_title'] )
+			wp_cache_add( $cache_bucket, ob_get_flush(), 'widget');
+
+		stats_extra( 'widget_view', 'authors' );
+	}
+
+	public function form( $instance ) {
+		$instance = wp_parse_args( $instance, array( 'title' => '', 'all' => false, 'avatar_size' => 48, 'number' => 0 ) );
+
+		?>
+		<p>
+			<label>
+				<?php _e( 'Title:' ); ?>
+				<input class="widefat" name="<?php echo $this->get_field_name( 'title' ); ?>" type="text" value="<?php echo esc_attr( $instance['title'] ); ?>" />
```

Comment:
In hindsight, it's probably needless, and would just waste cpu cycles on strings that have no user generated content.  It's all procedural IDs.